### PR TITLE
Do not attempt to jsonify if value is expression

### PR DESCRIPTION
**Pull request by gherceg**
## Technical Summary

We make use of Django [Func](https://docs.djangoproject.com/en/3.2/ref/models/expressions/#django.db.models.Func) expressions to execute [jsonb_set](https://github.com/dimagi/commcare-hq/blob/7ca7a082a0808afabe0e93beecca3ea845286531/corehq/sql_db/jsonops.py#L75-L75) in SQL (my understanding).

These expressions were impacted by a change to the SQLCompiler classes in Django, where `get_db_prep_save` is [now called](https://github.com/django/django/commit/0ff46591ac3010841c73fd26e0fef93995fedd99#diff-f58de2deaccecd2d53199c5ca29e3e1050ec2adb80fb057cdfc0b4e6accdf14fL1640) for all field values of a query, even if they are expressions. We don't actually want to jsonify an expression, so in our custom JsonAttrsField class where we define `get_prep_value`, which is eventually called by `get_db_prep_value`, we just return the value as is if it is of type Expression.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/util/jsonattrs.py [hunk 1]
```diff
@@ -100,6 +101,8 @@ def __init__(self, *args, builder, **kw):
         self.builder = builder
 
     def get_prep_value(self, value):
+        if isinstance(value, Expression):
```
Comment (millerdev) [inline corehq/util/jsonattrs.py:104]: Given [this line](https://github.com/django/django/blob/0ff46591ac3010841c73fd26e0fef93995fedd99/django/db/models/sql/compiler.py#L1618) just above the [code that you linked](https://github.com/django/django/commit/0ff46591ac3010841c73fd26e0fef93995fedd99#diff-f58de2deaccecd2d53199c5ca29e3e1050ec2adb80fb057cdfc0b4e6accdf14fL1640), I think we should use `hasattr(value, "resolve_expression")` here. [`resolve_expression` is an attribute of `BaseExpression`](https://github.com/django/django/blob/0ff46591ac3010841c73fd26e0fef93995fedd99/django/db/models/expressions.py#L258), which suggests there may be expression types that do not inherit from `Expression`.

**Review by millerdev — COMMENTED** [review]
Do you know if Django documented this change in compiler behavior? It looks like they did it to allow `Field`s to participate in expression resolution, which might be nice. But it also seems like a potentially breaking change for any field that implements `get_db_prep_save` or `get_prep_value`.

**Comment by gherceg** [comment]
It looks like this change was made as part of adding support for [psycopg3](https://code.djangoproject.com/ticket/33308), but I didn't see anything in release notes about changes to `get_prep_value` or anything.

**Comment by gherceg** [comment]
Oh actually I missed [this](https://docs.djangoproject.com/en/5.0/releases/4.2/#psycopg-3-support):
> Django now supports [psycopg](https://www.psycopg.org/psycopg3/) version 3.1.8 or higher. To update your code, install the [psycopg library](https://pypi.org/project/psycopg/), you don’t need to change the [ENGINE](https://docs.djangoproject.com/en/5.0/ref/settings/#std-setting-DATABASE-ENGINE) as django.db.backends.postgresql supports both libraries.
>
> Support for psycopg2 is likely to be deprecated and removed at some point in the future.
>
> Be aware that psycopg 3 introduces some breaking changes over psycopg2. As a consequence, you may need to make some changes to account for [differences from psycopg2](https://www.psycopg.org/psycopg3/docs/basic/from_pg2.html).

File: corehq/util/jsonattrs.py (hunk 1 above)
Comment (gherceg) [reply corehq/util/jsonattrs.py]: See 379cf00f21d14b34e2a46a9ce720bde6afb7f91b

**Review by millerdev — APPROVED** [review]

**Pull request merged**